### PR TITLE
Added generics to style guide

### PR DESCRIPTION
**Pull request by ghost**
Just added a section to cover generics and how deep we want to add the class types within arrays, dictionaries and sets.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -359,6 +359,27 @@ Nullability should always be used in header files, it's preferred in implementat
...
 Generics should be adopted wherever possible in header files - this helps document the code. It's optional in implementation files, though I've found it useful.
 
+When dealing with nested arrays, sets and dictionaries. Generics should look neat and be easily readable, therefore it is not recommended to go beyond displaying generics for the first level when arrays are nested within arrays etc.
+
+**Preferred:**
+
+```objc
+- (void)displayMyArray:(NSArray<NSSet *> *)myArray;
+- (void)displayMyDictionary:(NSDictionary<NSString *, NSArray *> *)myDictionary;
+- (void)displayMySet:(NSSet<NSArray <NSDictionary *> *> *)mySet;
```
Comment (nap-sam-dean) [inline README.md]: Doesn't this one break your rule of only going one deep?

Comment (ghost) [inline README.md]: Damn, after looking at so many generics for ten minutes I couldn't even make out my own code. I'll revise it now :)

**Comment by kerrmarin** [comment]
@nap-sam-dean Any update on this pull request?

**Comment by nap-sam-dean** [comment]
Sorry, thought I'd merged this ages ago!

**Pull request merged**